test/parallel/test-util-callbackify.js

```diff
@@ -0,0 +1,188 @@
+'use strict';
+const common = require('../common');
+
+// This test checks that the semantics of `util.callbackify` are as described in
+// the API docs
+
+const assert = require('assert');
+const { callbackify } = require('util');
+const { join } = require('path');
+const { execFile } = require('child_process');
+const fixtureDir = join(common.fixturesDir, 'uncaught-exceptions');
+const sentinelValues = [
+  'hello world',
+  null,
+  undefined,
+  { key: 'value' },
+  Symbol('I am a symbol'),
+  function ok() {},
+  ['array', 'with', 4, 'values'],
+  new Error('boo')
+];
+
+{
+  // Test that the resolution value is passed as second argument to callback
+  for (const sentinel of sentinelValues) {
+    async function asyncFn() {
+      return await Promise.resolve(sentinel);
+    }
+
+    const cbAsyncFn = callbackify(asyncFn);
+
+    cbAsyncFn(common.mustCall((err, ret) => {
+      assert.ifError(err);
+      assert.strictEqual(ret, sentinel);
+    }));
+
+    function promiseFn() {
+      return Promise.resolve(sentinel);
+    }
+
+    const cbPromiseFn = callbackify(promiseFn);
+
+    cbPromiseFn(common.mustCall((err, ret) => {
+      assert.ifError(err);
+      assert.strictEqual(ret, sentinel);
+    }));
+  }
+}
+
+{
+  // Test that rejection reason is passed as first argument to callback
+  for (const sentinel of sentinelValues) {
+    async function asyncFn() {
+      return await Promise.reject(sentinel);
+    }
+
+    const cbAsyncFn = callbackify(asyncFn);
+
+    cbAsyncFn(common.mustCall((err, ret) => {
+      assert.strictEqual(ret, undefined);
+      if (err instanceof Error) {
+        if ('reason' in err) {
+          assert.strictEqual(err.code, 'NULL_REJECTION');
+          assert.strictEqual(err.reason, sentinel);
+        } else {
+          assert.strictEqual(String(sentinel).endsWith(err.message), true);
+        }
+      } else {
+        assert.strictEqual(err, sentinel);
+      }
+    }));
+  }
+
+  for (const sentinel of sentinelValues) {
+    function promiseFn() {
+      return Promise.reject(sentinel);
+    }
+
+    const cbPromiseFn = callbackify(promiseFn);
+
+    cbPromiseFn(common.mustCall((err, ret) => {
+      assert.strictEqual(ret, undefined);
+      if (err instanceof Error) {
+        if ('reason' in err) {
+          assert.strictEqual(err.code, 'NULL_REJECTION');
+          assert.strictEqual(err.reason, sentinel);
```

Comment:
Maybe just leave `null` off the list, and have separate tests.

If not, I think this block is supposed to have an `assert.strictEqual(sentinel, null);` check?